packages/DatePicker/src/DatePicker.js

```diff
@@ -44,6 +44,9 @@ const propTypes = {
 
   /** Callback when date is selected or input. */
   onChange: PropTypes.func.isRequired,
+
+  /** Callback when input string is not correctly parsed as a date . */
+  onDateParseError: PropTypes.func,
```

Comment:
what about having `onError()` 
```
<DatePicker onError={({ input, errorMessage }) => {
   // do whatever with the error.
}}
```

I think onError is more reusable pattern across our components and we could communicate what kind of error is by passing an object describing the error.

Reply:
Sure but we already have a `hasError` prop (which is quite similar) in this component so I didn't want to change that too much at this point. I agree that `onError` would be preferable. Perhaps we can rename `hasError` prop to `hasConsumerError` or `hasExternalError`? @allison-c @nahumzs wdyt?

Reply:
`hasError` is a prop that our inputs (any kind) use, so I wouldn't suggest renaming that one. I guess you would want to manipulate that one when an event onError occur. So I think is safe to use onError. isn't?

Reply:
`hasError` will only describe if the input has a red border or red font as an example right?

Reply:
Ok yeah I just think it is quite easy for someone to confuse the two `onError` and `hasError` some wanted to be more declarative. I think the 'hasError` prop should really be on the `DatePicker.Input` component only and not on the `DatePicker` component? At least that would be less confusing?